lib/buffer.js

```diff
@@ -208,23 +209,23 @@ function _copy(source, target, targetStart, sourceStart, sourceEnd) {
   if (targetStart === undefined) {
     targetStart = 0;
   } else {
-    targetStart = toInteger(targetStart, 0);
+    targetStart = NumberIsInteger(targetStart) ? targetStart : toInteger(targetStart, 0);
```

Comment:
Maybe this optimization can be moved into toInteger() function to make the code cleaner?

Reply:
That would make it slower as V8 is not super good at inlining.